### ISSUE TITLE
Application Crashes on ios 9

### ISSUE THREAD

**Issue by leoo232** [paypal/PayPal-Cordova-Plugin#114]
HI, im doing an application with HTML5 and js. I have connected the paypal to to a Sndbox and payment were successfully been made on my iphone 4. But when installing on my friend's ipad mini 2 with ios 9, then hitting the button pay with paypal, it crashes the application. I'm using phonegap online build and have set the paypal version to 3.1.2 as advised on the webite.
can any one help plz.. thx

**Comment by bluk** [comment]
@leoo232 Can you try using version 3.1.16 for the Cordova plugin? There have been many updates to the mobile libraries since 3.1.2 (I assume that's the Cordova plugin version that you are using).

I assume you are looking at https://build.phonegap.com/plugins/1769 for the recommended version? We will try to get the version updated to the latest.

**Comment by leoo232** [comment]
Yes, its exactly at what im looking at on the phone gap site. done the necessary modification to put the version to 3.1.16 but got the following error:

plugin unsupported: com.paypal.cordova.mobilesdk @ 3.1.16

thx

**Comment by bluk** [comment]
It appears that public plugin submission has been disabled (only private is enabled) with the recommendation to [use Cordova plugins on npm](https://build.phonegap.com/plugins) so we can't update the version easily. Looking at the Phonegap Build documentation here: http://docs.build.phonegap.com/en_US/configuring_plugins.md.html#Plugins it seems you must add a `source` attribute to get new Cordova plugins on npm:

```
<gap:plugin name="com.paypal.cordova.mobilesdk" version="3.1.16" source="npm" />
```

Does that work?

**Comment by leoo232** [comment]
hi @bluk 

i have tried the version="3.1.16" source="npm" , but when it arrives on the page where there is the paypal button (<button id="buyNowBtn">) the app crashes on iphone 4. havent tested on ipad/ios9 yet.

do i need to update the paypal-mobile-js-helper.js - is there a new version, im using one that i found back in august 2015

thx

**Comment by romk1n** [comment]
@leoo232 the `paypal-mobile-js-helper.js` comes with a plugin make sure you don't have 2.

**Comment by leoo232** [comment]
@bluk i have one in my js folders referenced on the page with the button for paypal.. it was working with 3.1.2

**Comment by leoo232** [comment]
@bluk @romk1n do i need to remove the js reference now?

**Comment by bluk** [comment]
@leoo232 It depends on your project if you need to remove that specific reference. I would try to get @romk1n 's MyCordovaShop working ( https://github.com/romk1n/MyCordovaShop ) with the latest code and see if that works for you.

**Comment by leoo232** [comment]
@bluk I downloaded the MyCordovaShop from the link you gave and compiled it with my config file and it works... but still doesnt work with mine. tried removing the paypal-mobile-js-helper.js from the js folder as in the MyCordovaShop, still doent work. The pay with paypal wont call the paypal UI

My code is as follows

<html>
<head>
    <title>Concorde</title>
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta charset="utf-8">
    <link rel="stylesheet" href="http://maxcdn.bootstrapcdn.com/bootstrap/3.3.5/css/bootstrap.min.css">
    <link href="css/main.css" rel="stylesheet" />
<script type="text/javascript" src="cordova.js">
    <script type="text/javascript" src="js/paypal-mobile-js-helper.js"></script>
    <script type="text/javascript" src="js/book.js"></script>
    <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.3/jquery.min.js"></script>
    <script src="http://maxcdn.bootstrapcdn.com/bootstrap/3.3.5/js/bootstrap.min.js"></script>
    <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.0/jquery.min.js"></script>
    <script src="http://ajax.googleapis.com/ajax/libs/jquery/1.6/jquery.min.js" type="text/javascript"></script>
    <script src="http://ajax.googleapis.com/ajax/libs/jqueryui/1.8/jquery-ui.min.js" type="text/javascript"></script>
    <link href="http://ajax.googleapis.com/ajax/libs/jqueryui/1.8/themes/base/jquery-ui.css" rel="Stylesheet" type="text/css" />


    <script src="js/bootstrap.min.js"></script>

    <script type="text/javascript">
        $(document).ready(function () {

            var elem = document.getElementById("ex3");
            elem.value = localStorage.excurTotPrice;
            //document.getElementById("ex3").innerHTML = localStorage.excurTotPrice;
           
            document.getElementById("buyNowBtn").innerHTML = localStorage.PayPal;

        });
    </script>
</head>
<body style="background-color:#e6e6e6;">
    <div class="container-fluid">
        <br>
        <a href="Main.html"> <img src="images/logo.png" alt="Error in Image"></a>

        <br><br><br>
        <div class="col-xs-12">
            <label id="pmt"></label>
            <input class="form-control" id="ex3" type="text" readonly>
        </div>

    </div>
    <br /><br />
    <div class="col-xs-12">
        &nbsp; &nbsp;
        
    </div>
    <div class="col-xs-12">
        <div id="deviceready" class="blink">
            <p visibility="hidden" class="event listening"></p>
            <p visibility="hidden" class="event received"></p>
            <button id="buyNowBtn" class="btn btn-success" style="float:left; padding:5px 10px 5px 10px; background-color:#ff8000;text-decoration: none;border:1px solid #000; border-color: #6a6a6a; color: #ffffff"></button>
        </div>
    </div>
    <p>&nbsp;</p>

    <script>
        document.getElementById("pmt").innerHTML = localStorage.TotalPay;
    </script>
</body>
</html>

**Comment by leoo232** [comment]
<script type="text/javascript" src="cordova.js">
    <script type="text/javascript" src="js/paypal-mobile-js-helper.js"></script>
    <script type="text/javascript" src="js/book.js"></script>

**Comment by leoo232** [comment]
@bluk I downloaded the MyCordovaShop from the link you gave and compiled it with my config file and it works... but still doesnt work with mine. tried removing the paypal-mobile-js-helper.js from the js folder as in the MyCordovaShop, still doent work. The pay with paypal wont call the paypal UI

it also tried to put content of the index page of MyCordovaShop  inside my booking page. but still wont render anything when the button is being clicked. maybe there is some js conflicts inside.

Cleaning the codea bit. will update you with the outcome

thx guys

**Comment by leoo232** [comment]
@bluk @romk1n Tried version 3.1.17 on iphone 4. Reached paypal page but app closes immediately(crash)
 Please advise

thx

**Comment by leoo232** [comment]
android throws the following error. all built using online build

rror - Plugin error (you probably need to remove plugin files from your app): Fetching plugin "com.paypal.cordova.mobilesdk@3.1.16" via npm Installing "com.paypal.cordova.mobilesdk" for android Error during processing of action! Attempting to revert... Failed to install 'com.paypal.cordova.mobilesdk':Error: Uh oh! "/tmp/gimlet/569/1658517/www_android/project/cordova/plugins/com.paypal.cordova.mobilesdk/src/android/libs/arm64_v8a" not found! at Object.module.exports.common.copyFile (/home/ec2-user/.npm/lib/node_modules/pgb-plugman/node_modules/pgb-cordova-lib/src/plugman/platforms/common.js:38:40) at Object.module.exports.common.copyNewFile (/home/ec2-user/.npm/lib/node_modules/pgb-plugman/node_modules/pgb-cordova-lib/src/plugman/platforms/common.js:69:16) at Object.module.exports.source-file.install (/home/ec2-user/.npm/lib/node_modules/pgb-plugman/node_modules/pgb-cordova-lib/src/plugman/platforms/android.js:78:20) at installWrapper (/home/ec2-user/.npm/lib/node_modules/pgb-plugman/node_modules/pgb-cordova-lib/src/platforms/platforms.js:77:32) at Object.ActionStack.process (/home/ec2-user/.npm/lib/node_modules/pgb-plugman/node_modules/pgb-cordova-lib/src/plugman/util/action-stack.js:68:25) at handleInstall (/home/ec2-user/.npm/lib/node_modules/pgb-plugman/node_modules/pgb-cordova-lib/src/plugman/install.js:576:20) at /home/ec2-user/.npm/lib/node_modules/pgb-plugman/node_modules/pgb-cordova-lib/src/plugman/install.js:368:24 at _fulfilled (/home/ec2-user/.npm/lib/node_modules/pgb-plugman/node_modules/q/q.js:787:54) at self.promiseDispatch.done (/home/ec2-user/.npm/lib/node_modules/pgb-plugman/node_modules/q/q.js:816:30) at Promise.promise.promiseDispatch (/home/ec2-user/.npm/lib/node_modules/pgb-plugman/node_modules/q/q.js:749:13) Uh oh! "/tmp/gimlet/569/1658517/www_android/project/cordova/plugins/com.paypal.cordova.mobilesdk/src/android/libs/arm64_v8a" not found! - You can fix this here

**Comment by bluk** [comment]
For the Android error, you can try using version `3.1.18` (`3.1.17` should have fixed it and `3.1.18` has a bug fix for iOS). @romk1n made a fix for the wrong directory error.

For the iOS crash, is there anything in the console or anything else? I would really try starting with a simple page, especially on an iPhone 4. Older devices may stop your app because it runs out of available memory.

**Comment by leoo232** [comment]
@bluk got this when i tried 3.1.18

plugin unsupported: com.paypal.cordova.mobilesdk @ 3.1.18

**Comment by sanctus671** [comment]
is 3.1.18 tested and working on ios9? I had some issues with it crashing on ios9 too so hoping 3.1.18 has it fixed

**Comment by bluk** [comment]
@sanctus671 Yes, it should work. There was a network issue with iOS 7 that was fixed with `3.1.19`.

**Comment by leoo232** [comment]
hi all, 3.1.18 works on ios 9. tested on ipad mini 2

**Comment by bluk** [comment]
@leoo232 Great! I'll close this issue but if you have any other problems, please feel free to open a new issue.
